openshift/machine-config-operator
# move envtest helpers into framework package for easier reuse

**- What I did**

I moved the EnvTest setup and environment cleanup code into the `test/framework` package for easier reuse across the rest of the MCO codebase. Additionally, I refactored the Makefile as well as some of the test scripts to collect JUnit test reports in CI as well as a test coverage report.

**- How to verify it**

Because `test/e2e-bootstrap` is the only test suite that uses EnvTest, one can run `$ make bootstrap-e2e` or run the test target via `/test bootstrap-unit`. (For some unknown reason, we're not running `bootstrap-unit` on each PR, though it makes sense to because they're not overly resource intensive and execute quickly.)

For the JUnit and test coverage reports, open the links for the various Prow CI jobs and you should see them enumerated in the Test Results pane as well as a red / green graphic illustrating test coverage.

**- Description for the changelog**
Move EnvTest helpers into framework package for easier reuse
